Tribler/Core/Modules/restapi/channels_endpoint.py

```diff
@@ -52,15 +55,33 @@ class ChannelsSubscribedEndpoint(BaseChannelsEndpoint):
     }
     """
 
-    def __init__(self, session):
-        resource.Resource.__init__(self)
-        self.session = session
-        self.channel_db_handler = self.session.open_dbhandler(NTFY_CHANNELCAST)
-
     def render_GET(self, request):
         subscribed_channels_db = self.channel_db_handler.getMySubscribedChannels(includeDispsersy=True)
-        results_json = []
-        for channel in subscribed_channels_db:
-            results_json.append(self.convert_db_channel_to_json(channel))
-
+        results_json = [self.convert_db_channel_to_json(channel) for channel in subscribed_channels_db]
         return json.dumps({"subscribed": results_json})
+
+
+class ChannelsAllEndpoint(BaseChannelsEndpoint):
+    """
+    A GET request to this endpoint returns all channels in Tribler.
```

Comment:
All channels that Tribler has received from peers, you have no guarantee that you know all channels yet. @qstokkink is doing experiments with this stuff :D